geps/gep-2648/index.md

```diff
@@ -0,0 +1,415 @@
+
+## Direct Policy design rules
+
+With this example in mind, here are some rules for when a Policy is a Direct
+Policy:
+
+* The number or scope of objects is exactly _one_ object. No label selectors or
+  lists of `targetRef` are allowed.
+* The modifications to be made to the objects don’t have any transitive information -
+  that is, the modifications MUST only affect the single object that the targeted
+  metaresource is bound to, and MUST NOT have ramifications that flow beyond that
+  object. No attaching a Policy to a Gateway and affecting settings in Routes or
+  backends. If a Direct Attached Policy attaches to an object, it can only affect
+  properties _of that object_.
+* In terms of status, it SHOULD be reasonably easy for a user to understand that
+  everything is working - basically, as long as the targeted object exists, and
+  the modifications are valid, the metaresource is valid, and this should be
+  straightforward to communicate in one or two Conditions. The `status` stanza
+  in BackendTLSPolicy is an example of one of the recommended ways to achieve this.
+* Direct Policy Attachment SHOULD only be used to target objects in the same
+  namespace as the Policy object. Allowing cross-namespace references brings in
+  significant security concerns, and/or difficulties about merging cross-namespace
+  policy objects. Notably, Mesh use cases may need to do something like this for
+  consumer policies, but in general, Policy objects that modify the behavior of
+  things outside their own namespace should be avoided unless it uses a handshake
+  of some sort, where the things outside the namespace can opt–out of the behavior.
+  (Notably, this is the design that we used for ReferenceGrant).
+
+## Apply Policies to Sections of a Resource
+
+The `sectionName` field of `targetRef` can be used to target a specific section of other resources:
+
+* Service.Ports.Name
+* Gateway.Listners.Name
+* HTTPRoute.Rules.Name (once they are added in GEP-995, PR at https://github.com/kubernetes-sigs/gateway-api/pull/2593)
+
+For example, the RetryPolicy below applies to a RouteRule inside an HTTPRoute.
+(or rather, it will when GEP-995 merges).
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-app-1
+  labels:
+    app: foo
+spec:
+  hostnames:
+  - "foo.com"
+  rules:
+  - name: bar
+    matches:
+    - path:
+        type: Prefix
+        value: /bar
+    backendRefs:
+    - name: my-service1
+      port: 8080
+---
+apiVersion: networking.acme.io/v1alpha2
+kind: RetryPolicy
+metadata:
+  name: foo
+spec:
+  maxRetries: 5
+  targetRef:
+    name: http-app-1
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    sectionName: bar
+```
+
+If a `sectionName` is specified, but does not exist on the targeted object, the
+Policy must fail to attach, and the policy implementation should record a
+`resolvedRefs` or similar Condition in the Policy's status.
+
```

Comment:
Could you add a brief description of what happens when more than one object matches the sectionName in the targetRef?

Reply:
I don't think this should be possible/allowed - I believe `targetRef` is only intended to select a single object as defined by a unique group/kind/namespace/name/sectionName.

I think https://github.com/kubernetes-sigs/gateway-api/pull/2813#discussion_r1552455260 may propose a possible path forward for your use case though?